File: examples_ios/MyApp/Sources/ViewModels/Models/SurveyModel.swift
import Foundation
import MobileSdk

class SurveyModel {
    private(set) var survey: Survey
    var selectedLanguage: SurveyLanguage
    private(set) var configs: SurveyConfigs
    
    convenience init(serverId: String, surveyId: String) throws {
        guard let survey = try SurveySDK.getSurvey(serverId: serverId, surveyId: surveyId) else {
            throw AppError.general("Survey is not available")
        }
        
        try self.init(survey: survey)
    }
    
    init(survey: Survey) throws {
        self.survey = survey
        self.selectedLanguage = try survey.getDefaultLanguage()
        self.configs = SurveyConfigs(serverId: survey.serverId, surveyId: survey.surveyId)
    }
    
    func update(survey: Survey) {
        self.survey = survey
    }
    
    func getRespondentValues() -> [String: String] {
        var values = [String: String]()
        let raw = configs.respondentValue
        for pair in raw.split(separator: ";") {
            let keyValue = pair.split(separator: "=")
            if keyValue.count == 2 {
                let key = String(keyValue[0])
                let value = String(keyValue[1])
                values[key] = value
            }
        }
        return values
    }
}

File: examples_android/app/src/main/java/com/confirmit/testsdkapp/viewmodels/models/SurveyModel.kt
package com.confirmit.testsdkapp.viewmodels.models

import com.confirmit.mobilesdk.SurveySDK
import com.confirmit.mobilesdk.database.externals.Survey
import com.confirmit.mobilesdk.database.externals.SurveyLanguage
import com.confirmit.testsdkapp.AppExceptions
import com.confirmit.testsdkapp.SurveyConfigs
import java.io.Serializable

class SurveyModel : Serializable {
    var survey: Survey
        private set
    var selectedLanguage: SurveyLanguage
    var configs: SurveyConfigs
        private set

    constructor(serverId: String, surveyId: String) {
        val survey = SurveySDK.getSurvey(serverId, surveyId)
                ?: throw AppExceptions("Survey is not available")
        this.survey = survey
        this.selectedLanguage = survey.getDefaultLanguage()
        this.configs = SurveyConfigs(survey.serverId, survey.surveyId)
    }

    constructor(survey: Survey) {
        this.survey = survey
        this.selectedLanguage = survey.getDefaultLanguage()
        this.configs = SurveyConfigs(survey.serverId, survey.surveyId)
    }

    fun update(survey: Survey) {
        this.survey = survey
    }

    fun getRespondentValues(): Map<String, String> {
        val values = mutableMapOf<String, String>()
        val raw = configs.respondentValue
        raw.split(";").forEach {
            val keyValue = it.split("=")
            if (keyValue.size == 2) {
                val key = keyValue[0]
                val value = keyValue[1]
                values[key] = value
            }
        }
        return values
    }

}